File: bindings/python/test_wareki.py
import wareki
import datetime
import pytest

def test_to_wareki_normal():
    # 2026-02-23 -> 令和8年
    w = wareki.to_wareki(2026, 2, 23)
    assert w.era_name == "令和"
    assert w.year == 8
    assert str(w) == "令和8年"

def test_to_wareki_heisei():
    # 1989-01-08 -> 平成1年
    w = wareki.to_wareki(1989, 1, 8)
    assert w.era_name == "平成"
    assert w.year == 1

def test_to_wareki_error_before_meiji():
    # 明治以前 (1868-01-24)
    with pytest.raises(ValueError, match="Date is out of supported range"):
        wareki.to_wareki(1868, 1, 24)

def test_from_wareki_normal():
    # 令和8年2月23日 -> 2026-02-23
    d = wareki.from_wareki("令和", 8, 2, 23)
    assert d == datetime.date(2026, 2, 23)

def test_from_wareki_abbreviations():
    assert wareki.from_wareki("令", 8, 2, 23) == datetime.date(2026, 2, 23)
    assert wareki.from_wareki("r", 8, 2, 23) == datetime.date(2026, 2, 23)
    assert wareki.from_wareki("R", 8, 2, 23) == datetime.date(2026, 2, 23)

def test_from_wareki_leap_year():
    # 令和6年2月29日(2024年)
    d = wareki.from_wareki("令和", 6, 2, 29)
    assert d == datetime.date(2024, 2, 29)

def test_from_wareki_invalid_date():
    # 令和5年は閏年ではないためエラー
    with pytest.raises(ValueError):
        wareki.from_wareki("令和", 5, 2, 29)

File: bindings/node/test_wareki.js
const assert = require('assert');
const wareki = require('./index.js');

function runTests() {
    console.log("Running Node.js tests...");

    // to_wareki normal
    const w1 = wareki.toWareki(2026, 2, 23);
    assert.strictEqual(w1.eraName, "令和");
    assert.strictEqual(w1.year, 8);

    const w2 = wareki.toWareki(1989, 1, 8);
    assert.strictEqual(w2.eraName, "平成");
    assert.strictEqual(w2.year, 1);

    // to_wareki string representation test
    assert.strictEqual(`${w1.eraName}${w1.year}年`, "令和8年");

    // from_wareki normal
    const d1 = wareki.fromWareki("令和", 8, 2, 23);
    assert.strictEqual(d1, "2026-02-23");

    // from_wareki abbreviations
    assert.strictEqual(wareki.fromWareki("令", 8, 2, 23), "2026-02-23");
    assert.strictEqual(wareki.fromWareki("r", 8, 2, 23), "2026-02-23");
    assert.strictEqual(wareki.fromWareki("R", 8, 2, 23), "2026-02-23");

    // leap year
    const leapDate = wareki.fromWareki("令和", 6, 2, 29);
    assert.strictEqual(leapDate, "2024-02-29");

    // invalid leap year
    assert.throws(() => {
        wareki.fromWareki("令和", 5, 2, 29);
    }, { code: 'InvalidArg' });

    // out of range before Meiji
    assert.throws(() => {
        wareki.toWareki(1868, 1, 24);
    }, { code: 'InvalidArg' });

    console.log("All Node.js tests passed! \u2705");
}

runTests();
